pkg/stanza/operator/helper/encoding.go

```diff
@@ -45,28 +45,27 @@ func (c EncodingConfig) Build() (Encoding, error) {
 	}
 
 	return Encoding{
-		Encoding:     enc,
-		decodeBuffer: make([]byte, 1<<12),
-		decoder:      enc.NewDecoder(),
+		Encoding: enc,
+		decoder:  enc.NewDecoder(),
 	}, nil
 }
 
 type Encoding struct {
-	Encoding     encoding.Encoding
-	decoder      *encoding.Decoder
-	decodeBuffer []byte
+	Encoding encoding.Encoding
+	decoder  *encoding.Decoder
 }
 
 // Decode converts the bytes in msgBuf to utf-8 from the configured encoding
 func (e *Encoding) Decode(msgBuf []byte) ([]byte, error) {
```

Comment:
Another option is to allow the user to pass in the output buffer.
That would be a perf improvement since long lived goroutines that call `Decode()` could reuse the same output buffer and still avoid sharing an output buffer by different goroutines.

Reply:
Could also just build the encoding for each goroutine.

Reply:
Passing in the output buffer helps so long as there isn't any resizing happening.

Reply:
Can have it return the decode buffer as well.